
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [2]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
# number of callbacks for balck-sounding names
sum(data[data.race=='b'].call)

157.0

In [5]:
data.head(5)

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## 1. What test is appropriate for this problem? Does CLT apply?
I will do hypothesis testing comparing sample proportions. The populations will be black sounding names versus white sounding names. Since the values for callbacks are either 0 or 1, I can use Bernoulli distributions. The CLT applies to the Bernoulli distribution and my sample sizes are large enough.

## 2. What are the null and alternate hypotheses?

Null Hypothesis H0: Race has no significant impact on the rate of callbacks for resumes. 

Alternate Hypothesis H1: Race has a significant impact on the rate of callbacks for resumes.

## 3. Compute margin of error, confidence interval, and p-value.

In [13]:
from __future__ import division
n1 = len(data[(data.race == 'b')])
p1 = len(data[(data.race == 'b') & (data.call == 1)])/n1
print("Proportion of b race getting a callback:")
print(p1)
n2 = len(data[(data.race == 'w')])
p2 = len(data[(data.race == 'w') & (data.call == 1)])/n2
print("Proportion of w race getting a callback:")
print(p2)


Proportion of b race getting a callback:
0.064476386037
Proportion of w race getting a callback:
0.0965092402464


In [16]:
import math
prop_diff = p2 - p1
print("Difference in proportions:")
print(prop_diff)
std_diff = math.sqrt((p1*(1-p1)/n1) + p2*(1-p2)/n2)
print("STD of diff:")
print(std_diff)


Difference in proportions:
0.0320328542094
STD of diff:
0.00778337058668


In [17]:
# Assuming 95% Confidence Interval find the critical z-value
z_critical = stats.norm.ppf(q = 0.975)
print("z-critical value:")              # Check the z-critical value
print(z_critical)

margin_of_error = z_critical * (std_diff)
print("margin of error:")
print(margin_of_error*100)

confidence_interval = (prop_diff - margin_of_error,
                       prop_diff + margin_of_error)  

print("Confidence interval:")
print(confidence_interval)
        

z-critical value:
1.95996398454
margin of error:
1.52551260282
Confidence interval:
(0.016777728181230755, 0.047287980237660412)


Assuming null hypothesis is true: P1-P0 = 0

In [28]:
import scipy 
from scipy import stats
from scipy.stats import norm
#Assuming no difference in w and b race, finding the proportion p who received call (1)
p_h0 = (len(data[(data.race == 'b') & (data.call == 1)]) + len(data[(data.race == 'w') & (data.call == 1)]))/(n1+n2)
print("Sample proportion assuming H0 is true:")
print(p_h0)
std_h0 = math.sqrt((p_h0*(1-p_h0)/n1)+(p_h0*(1-p_h0)/n2))
print("Sample std assuming H0 is true:")
print(std_h0)
z_h0 = prop_diff/std_h0
print("z-stat assuming H0 is true:")
print(z_h0)
if(z_h0 > z_critical):
    print("reject H0. There is significant difference")
else:
    print("Accept H0. No significant difference")
    
p_value = scipy.stats.norm.sf(abs(z_h0)) 
print ("p-value:", p_value)

Sample proportion assuming H0 is true:
0.0804928131417
Sample std assuming H0 is true:
0.00779689403617
z-stat assuming H0 is true:
4.10841215243
reject H0. There is significant difference
('p-value:', 1.9919434187925383e-05)


## 4. Discuss statistical significance.
There is only a .00199% chance that the difference in the proportions is .032 assuming that the null hypothesis is true. This is way below the 5% significance level and hence I reject the null hypothesis. 

In conclusion, race has a significant impact on the rate of callbacks for resumes. 